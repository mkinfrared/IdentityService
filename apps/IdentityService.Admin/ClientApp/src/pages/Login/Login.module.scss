@use "shared/styles/common" as c;

.Login {
}
