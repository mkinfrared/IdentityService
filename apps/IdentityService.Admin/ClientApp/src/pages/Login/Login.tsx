import { Button } from "@identity-service/ui";
import { memo, useCallback } from "react";

import userManager from "utils/userManager";

import css from "./Login.module.scss";

const Login = () => {
  const handleClick = useCallback(() => {
    userManager.signinRedirect();
  }, []);

  return (
    <div className={css.Login} data-testid="Login">
      <Button onClick={handleClick}>Login</Button>
    </div>
  );
};

export { Login };

export default memo(Login);
