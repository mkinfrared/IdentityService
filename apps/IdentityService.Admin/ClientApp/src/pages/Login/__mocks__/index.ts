export { default } from "./Login.mock";
