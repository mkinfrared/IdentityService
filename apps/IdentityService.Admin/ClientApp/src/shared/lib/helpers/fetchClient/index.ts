export { FetchClient } from "./fetchClient";

export { FetchError, isFetchError } from "./fetchError";
