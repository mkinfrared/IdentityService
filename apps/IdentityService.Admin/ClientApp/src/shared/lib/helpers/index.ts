export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { FetchClient, FetchError, isFetchError } from "./fetchClient";

export { lazyImport } from "./lazyImport";

export { userManager } from "./userManager";
