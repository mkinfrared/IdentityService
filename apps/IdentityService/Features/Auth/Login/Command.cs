using System.ComponentModel.DataAnnotations;

using MediatR;

using Microsoft.AspNetCore.Identity;

namespace IdentityService.Features.Auth.Login;

public partial class Login
{
    public class Command : IRequest<SignInResult>
    {
        public Command(string username, string password, string returnUrl)
        {
            Username = username;
            Password = password;
            ReturnUrl = returnUrl;
        }

        [Required]
        public string Username { get; }

        [Required]
        public string Password { get; }

        [Required]
        public string ReturnUrl { get; }
    }
}
