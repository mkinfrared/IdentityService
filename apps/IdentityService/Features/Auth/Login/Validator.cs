using FluentValidation;

namespace IdentityService.Features.Auth.Login;

public partial class Login
{
  public class Validator : AbstractValidator<Command>
  {
    public Validator()
    {
      RuleFor(command => command.Username)
        .NotEmpty();

      RuleFor(command => command.Password)
        .NotEmpty();

      RuleFor(command => command.ReturnUrl)
        .NotEmpty();
    }
  }
}
