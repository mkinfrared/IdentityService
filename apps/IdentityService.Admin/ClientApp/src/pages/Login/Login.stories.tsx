import { Meta, Story } from "@storybook/react";

import { Login } from "./Login";
import { LoginProps } from "./Login.type";
import css from "./Story.module.scss";

export default {
  title: "PAGES/Login",
  component: Login,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<LoginProps> = () => (
  <div className={css.Story}>
    <Login />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
