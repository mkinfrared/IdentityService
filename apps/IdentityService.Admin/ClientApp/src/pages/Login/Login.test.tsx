import { fireEvent, render } from "@testing-library/react";

import { userManager } from "shared/lib/helpers";

import Login from "./Login";

jest.mock("shared/lib/helpers");

describe("<Login />", () => {
  const userManagerMock = jest.mocked(userManager);
  const Component = <Login />;

  it("should be defined", () => {
    expect(Login).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Login");

    expect(element).toBeDefined();
  });

  it("should call 'signinRedirect' on userManager when button is clicked", () => {
    const { getByTestId } = render(Component);
    const button = getByTestId("Button");

    fireEvent.click(button);

    expect(userManagerMock.signinRedirect).toHaveBeenCalled();

    expect(userManagerMock.signinRedirect).toHaveBeenCalledTimes(1);
  });
});
