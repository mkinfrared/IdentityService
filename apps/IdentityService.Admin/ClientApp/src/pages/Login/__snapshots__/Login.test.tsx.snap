// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match the snapshot 1`] = `
<div>
  <div
    class="Login"
    data-testid="Login"
  >
    <button
      class="Button-module_Button__0OJEu Button-module_contained__B-47X"
      data-testid="Button"
      type="button"
    >
      <svg
        class="Button-module_spinner__kVZYq"
        fill="#fff"
        viewBox="0 0 135 140"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="120"
          rx="6"
          width="15"
          y="10"
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height="120"
          rx="6"
          width="15"
          x="30"
          y="10"
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height="140"
          rx="6"
          width="15"
          x="60"
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height="120"
          rx="6"
          width="15"
          x="90"
          y="10"
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height="120"
          rx="6"
          width="15"
          x="120"
          y="10"
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
      <span
        class="Button-module_content__PpYVU"
      >
        Login
      </span>
    </button>
  </div>
</div>
`;
